### PR TITLE
Composability updates for Splunk On-Call plugin

### PR DESCRIPTION
**Pull request by timbonicus**
## Hey, I just made a Pull Request!

Updates `plugin-splunk-on-call` to the new [composability system](https://backstage.io/docs/plugins/composability#porting-existing-plugins).

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/splunk-on-call/src/index.ts [hunk 1]
```diff
@@ -14,14 +14,12 @@
...
+  SplunkOnCallCard,
   splunkOnCallPlugin,
   splunkOnCallPlugin as plugin,
   SplunkOnCallPage,
 } from './plugin';
-export {
-  isPluginApplicableToEntity,
-  SplunkOnCallCard,
-} from './components/SplunkOnCallCard';
+export { isSplunkOnCallAvailable } from './components/EntitySplunkOnCallCard';
```
Comment (IlllIIIIlIIllIl) [inline plugins/splunk-on-call/src/index.ts:22]: Does this also need to include `EntitySplunkOnCallCard`?

File: plugins/splunk-on-call/src/plugin.ts [hunk 2]
```diff
@@ -49,3 +50,14 @@ export const SplunkOnCallPage = splunkOnCallPlugin.provide(
     mountPoint: rootRouteRef,
   }),
 );
+
+export const SplunkOnCallCard = splunkOnCallPlugin.provide(
```
Comment (freben) [inline plugins/splunk-on-call/src/plugin.ts:54]: These are commonly prefixed with the word `Entity` to signify that it's meant to be used in the catalog entity view.

File: plugins/splunk-on-call/src/index.ts (hunk 1 above)
Comment (freben) [reply plugins/splunk-on-call/src/index.ts:22]: I think it shouldn't; the one exported from the plugin above should be renamed to `EntitySplunkOnCallCard` instead.

File: plugins/splunk-on-call/src/plugin.ts (hunk 2 above)
Comment (timbonicus) [reply plugins/splunk-on-call/src/plugin.ts]: Right, renamed the actual file but didn't update the export here. Fixed 👍

**Review by freben — APPROVED** [review]

**Pull request merged**